crates/recursor/src/recursor.rs

```diff
@@ -266,8 +301,106 @@ impl Recursor {
         request_time: Instant,
         query_has_dnssec_ok: bool,
     ) -> Result<Lookup, Error> {
-        self.handle
-            .resolve(query, request_time, query_has_dnssec_ok)
-            .await
+        match &self.either {
+            RecursorEither::NonValidating(handle) => {
+                handle
+                    .resolve(query, request_time, query_has_dnssec_ok)
+                    .await
+            }
+
+            #[cfg(feature = "dnssec")]
+            RecursorEither::Validating {
+                handle,
+                record_cache,
+            } => {
+                let mut options = DnsRequestOptions::default();
+                options.use_edns = true;
+                // a validating recursor must be security aware
+                options.edns_set_dnssec_ok = true;
+
+                let response = handle.lookup(query.clone(), options).first_answer().await?;
+                let lookup = common::insert_response(
+                    response,
+                    None,
+                    record_cache,
+                    query.clone(),
+                    request_time,
+                )?;
+                Ok(common::maybe_strip_dnssec_records(
+                    query_has_dnssec_ok,
+                    lookup,
+                    query,
+                ))
+            }
+        }
+    }
+}
+
+enum RecursorEither {
+    NonValidating(RecursorDnsHandle),
```

Comment:
Why doesn't the non-validating variant need a cache?

I dislike `Either` naming because there is no semantic meaning. Can you come up with a better name for this?